dulwich/refs.py

```diff
@@ -233,7 +233,7 @@ def subkeys(self, base):
                 keys.add(refname[base_len:])
         return keys
 
-    def as_dict(self, base=None):
+    def as_dict(self, base=None) -> dict[bytes, bytes]:
```

Comment:
Please use Ref and ObjectID here - we're migrating to those types rather than the generic "bytes"